WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -703,6 +787,12 @@ - (void)deletePostsInExcessOfMaxAllowedForTopic:(ReaderAbstractTopic *)topic
         DDLogInfo(@"Deleting ReaderPost: %@", post.postTitle);
         [self.managedObjectContext deleteObject:post];
     }
+
+    // If the last remaining post is a gap marker, remove it.
+    ReaderPost *lastPost = [posts objectAtIndex:ReaderPostServiceMaxPosts - 1];
```

Comment:
(Maybe) a failsafe, just in case `posts` contain less than `ReaderPostServiceMaxPosts` ?